src/dynamics/initial_conditions.jl

```diff
@@ -313,20 +317,31 @@ function initialize!(
     initial_conditions::RossbyHaurwitzWave,
     model::AbstractModel,
```

Comment:
You have different options here: Either you constraint this method here to `model::ShallowWater` as you're now defining something that only really makes sense in the shallow water model. Imagine now someone wants to use the Rossby-Haurwitz initial conditions in the primitive equation model, there `pres` has a different meaning (logarithm of surface pressure)